File: csharp/Resources/OtelResource.cs
using System.Net.Http;
using System.Threading.Tasks;
using Newtonsoft.Json.Linq;

namespace Authdog
{
    /// <summary>
    /// OpenTelemetry export Wave 3 operations
    /// </summary>
    public class OtelResource
    {
        private readonly AuthdogClient _client;

        public OtelResource(AuthdogClient client)
        {
            _client = client;
        }

        public Task<JObject> ExportLogsAsync(object body) =>
            _client.RequestAsync<JObject>(HttpMethod.Post, "/v1/logs", body);

        public JObject ExportLogs(object body) =>
            ExportLogsAsync(body).GetAwaiter().GetResult();

        public Task<JObject> ExportMetricsAsync(object body) =>
            _client.RequestAsync<JObject>(HttpMethod.Post, "/v1/metrics", body);

        public JObject ExportMetrics(object body) =>
            ExportMetricsAsync(body).GetAwaiter().GetResult();

        public Task<JObject> ExportTracesAsync(object body) =>
            _client.RequestAsync<JObject>(HttpMethod.Post, "/v1/traces", body);

        public JObject ExportTraces(object body) =>
            ExportTracesAsync(body).GetAwaiter().GetResult();

        public Task<JObject> ExportLogsPrefixedAsync(object body) =>
            _client.RequestAsync<JObject>(HttpMethod.Post, "/v1/otel/v1/logs", body);

        public JObject ExportLogsPrefixed(object body) =>
            ExportLogsPrefixedAsync(body).GetAwaiter().GetResult();

        public Task<JObject> ExportMetricsPrefixedAsync(object body) =>
            _client.RequestAsync<JObject>(HttpMethod.Post, "/v1/otel/v1/metrics", body);

        public JObject ExportMetricsPrefixed(object body) =>
            ExportMetricsPrefixedAsync(body).GetAwaiter().GetResult();

        public Task<JObject> ExportTracesPrefixedAsync(object body) =>
            _client.RequestAsync<JObject>(HttpMethod.Post, "/v1/otel/v1/traces", body);

        public JObject ExportTracesPrefixed(object body) =>
            ExportTracesPrefixedAsync(body).GetAwaiter().GetResult();
    }
}

File: java/src/main/java/com/authdog/resources/OtelResource.java
package com.authdog.resources;

import com.authdog.AuthdogClient;
import com.authdog.exceptions.ApiException;
import com.authdog.exceptions.AuthenticationException;
import com.fasterxml.jackson.databind.JsonNode;

/**
 * OpenTelemetry export operations.
 */
public final class OtelResource {
    /**
     * Authdog client.
     */
    private final AuthdogClient client;

    /**
     * Create an OTEL helper.
     * @param clientParam Authdog client
     */
    public OtelResource(final AuthdogClient clientParam) {
        this.client = clientParam;
    }

    /**
     * Export logs to the unprefixed OTEL path.
     * @param body request body
     * @return export envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode exportLogs(final Object body)
            throws AuthenticationException, ApiException {
        return client.request("POST", "/v1/logs", body, null,
                JsonNode.class);
    }

    /**
     * Export metrics to the unprefixed OTEL path.
     * @param body request body
     * @return export envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode exportMetrics(final Object body)
            throws AuthenticationException, ApiException {
        return client.request("POST", "/v1/metrics", body, null,
                JsonNode.class);
    }

    /**
     * Export traces to the unprefixed OTEL path.
     * @param body request body
     * @return export envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode exportTraces(final Object body)
            throws AuthenticationException, ApiException {
        return client.request("POST", "/v1/traces", body, null,
                JsonNode.class);
    }

    /**
     * Export logs to the prefixed OTEL path.
     * @param body request body
     * @return export envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode exportLogsPrefixed(final Object body)
            throws AuthenticationException, ApiException {
        return client.request("POST", "/v1/otel/v1/logs", body, null,
                JsonNode.class);
    }

    /**
     * Export metrics to the prefixed OTEL path.
     * @param body request body
     * @return export envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode exportMetricsPrefixed(final Object body)
            throws AuthenticationException, ApiException {
        return client.request("POST", "/v1/otel/v1/metrics", body, null,
                JsonNode.class);
    }

    /**
     * Export traces to the prefixed OTEL path.
     * @param body request body
     * @return export envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode exportTracesPrefixed(final Object body)
            throws AuthenticationException, ApiException {
        return client.request("POST", "/v1/otel/v1/traces", body, null,
                JsonNode.class);
    }
}
